### PR TITLE
exposition: allow parquet conversion from file handles

### PR DESCRIPTION
**Pull request by mihirn**

### CODE REVIEW

File: metriken-exposition/src/convert.rs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    /// Converts a file with metrics in msgpack format to a parquet file.
+    /// Input and putput are open file handles.
+    /// If successful, returns the number of rows written out to the parquet
+    /// file.
+    pub fn convert_file_handle(
+        self,
+        reader: impl Read + Seek,
+        writer: impl Write + Send,
+    ) -> Result<i64, ParquetError> {
```
Comment (swlynch99) [inline metriken-exposition/src/convert.rs:53]: It's weird to me that this is an `i64`. Is there ever going to be a file with a negative number of rows?

File: metriken-exposition/src/lib.rs [hunk 2]
```diff
@@ -5,13 +5,15 @@
...
 pub use histogram::Snapshot as HistogramSnapshot;
 
+#[cfg(all(feature = "serde", feature = "msgpack", feature = "parquet"))]
+mod convert;
 #[cfg(feature = "parquet")]
 mod parquet;
 mod snapshot;
 mod snapshotter;
-#[cfg(all(feature = "serde", feature = "msgpack", feature = "parquet"))]
-pub mod util;
```
Comment (swlynch99) [inline metriken-exposition/src/lib.rs:13]: This is a breaking change - you need to bump the version number to `0.3.0`

File: metriken-exposition/src/lib.rs [hunk 3]
```diff
@@ -5,13 +5,15 @@
...
 
+#[cfg(all(feature = "serde", feature = "msgpack", feature = "parquet"))]
```
Comment (swlynch99) [inline metriken-exposition/src/lib.rs:15]: We should have one user-facing feature that covers all of this.

**Review by swlynch99 — CHANGES_REQUESTED** [review]
The actual code LGTM. I have one blocking issue and then a few minor issues and questions that aren't mandatory to fix.

Blocking issue:
- The `util` module is public so removing it is a breaking change and needs a major version bump.

Minor issues:
- The number of rows being an `i64` is weird to me. Is there a reason you've picked `i64` instead of `u64`?
- The fact that users need to enable all of the individual dependency features to get at the conversion functionality isn't great.

   I'd suggest creating one top-level feature (maybe `parquet-conversion`?) and having that enable the msgpack, parquet, and serde features.

File: metriken-exposition/src/convert.rs (hunk 1 above)
Comment (mihirn) [reply metriken-exposition/src/convert.rs:53]: Only to reflect whatever the underlying type from the parquet library is: https://github.com/apache/arrow-rs/blob/master/parquet/src/format.rs#L4916

I don't have a preference here -- it isn't clear to me whether returning anything is particularly useful vs just a `Result<(), ParquetError>`.

Comment (swlynch99) [reply metriken-exposition/src/convert.rs:53]: That seems fair to me. If we find that it becomes an issue then we can change it in the future.

**Review by swlynch99 — APPROVED** [review]
With the version bump this LGTM. I leave it up to you if you want to combine the features together.

You're free to merge once CI passes. Note that you'll need to commit an up-to-date Cargo.lock to get all the checks to pass.

**Pull request merged**